src/Chapter3.hs

```diff
@@ -971,6 +1079,30 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Monday
+             | Tuesday
+             | Wednesday
+             | Thursday
+             | Friday
+             | Saturday
+             | Sunday
+             deriving (Ord, Eq, Show, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Saturday = True
+isWeekend Sunday   = True
+isWeekend _        = False
+
+nextDay :: Weekday -> Weekday
+nextDay Sunday = Monday
+nextDay d = succ d
+
+daysToParty :: Weekday -> Int
+daysToParty Friday = 0
+daysToParty d
+    | fromEnum d < fromEnum Friday = fromEnum Friday - fromEnum d
+    | otherwise = abs (fromEnum Friday) - fromEnum d + fromEnum Friday
```

Comment:
You are using `fromEnum Friday` a lot here, so maybe it could be a good idea to put it into a separate var.

Also, I guess, you can combine these cases together with the usage of `mod` 😉